ticketbuyer/mempool.go

```diff
@@ -69,16 +69,22 @@ func (t *TicketPurchaser) ownTicketsInMempool() (int, error) {
 	// next block attached yet.
 	var curStakeInfo *wallet.StakeInfoData
 	var err error
+out:
 	for i := 0; i < stakeInfoReqTries; i++ {
-		curStakeInfo, err = t.wallet.StakeInfo()
-		if err != nil {
-			log.Tracef("Failed to fetch stake information "+
-				"on attempt %v: %v", i, err.Error())
-			time.Sleep(stakeInfoReqTryDelay)
-			continue
-		}
-		if err == nil {
-			break
+		select {
+		case <-t.quit:
+			return 0, nil
+		default:
+			curStakeInfo, err = t.wallet.StakeInfo()
```

Comment:
Might want to also consider propagating some sort of error to indicate a timeout instead of pretending it completed successfully and returning a bogus result.